text/000-sfc-and-template-import-primitives.md

```diff
@@ -0,0 +1,679 @@
+- Start Date: (fill me in with today's date, YYYY-MM-DD)
+- Relevant Team(s): Ember.js
+- RFC PR: (after opening the RFC PR, update this with a link to it and update the file name)
+- Tracking: (leave this empty)
+
+# SFC & Template Import Primitives
+
+## Summary
+
+Expose low-level primitives for associating templates with component classes
+and customizing a template's ambient scope.
+
+These primitives unlock experimentation, allowing addons to provide
+highly-requested features (such as single-file components) via stable, public
+API.
+
+## Motivation
+
+This proposal is intended to unlock experimentation around two
+highly-requested features:
+
+1. Single-file components.
+2. Module imports in component templates.
+
+Although exploring these features is the primary motivation, an important
+benefit of stabilizing low-level APIs is that they enable the Ember community
+to experiment with new, unexpected ideas.
+
+### Single-File Components
+
+In Ember components today, JavaScript code lives in a `.js` file and template
+code lives in a separate `.hbs` file. Juggling between these two files adds
+friction to the developer experience.
+
+Template and JavaScript code are inherently coupled, and changes to one are
+often accompanied by changes to the other. Separating them provides little
+value in terms of improving reusability or composability.
+
+Other component APIs eliminate this friction in different ways. React uses
+JSX, which produces JavaScript values:
+
+```js
+// MyComponent.jsx
+
+export default class extends Component {
+  state = { name: 'World' };
+
+  render() {
+    return <div>Hello {this.state.name}!</div>
+  }
+}
+```
+
+Vue has an optional [`.vue` single-file component (SFC)
+format](https://vuejs.org/v2/guide/single-file-components.html):
+
+```html
+<!-- MyComponent.vue -->
+
+<template>
+  <div>Hello {{name}}!</div>
+</template>
+
+<script>
+export default {
+  data() {
+    return {
+      name: 'World'
+    }
+  }
+}
+</script>
+```
```

Comment:
Is it worth also calling out how Angular utilizes template literals for inline templates?

```js
    import { Component } from '@angular/core';
     
    @Component({
      selector: 'app-root',
      template: `
        <h1>Hello {{name}}</h1>
      `
    })
    export class AppComponent {
      name = 'World';
    }
```

Reply:
only if we're using it as an example of something we'd never do...In all seriousness though, It may be better to leave it out.